torch/_refs/__init__.py

```diff
@@ -822,6 +822,33 @@ def add(
     supports_lhs_python_scalar=False,
 )
 
+
+def _pow(
+    a: Union[TensorLikeType, NumberType],
+    b: Union[TensorLikeType, NumberType],
+) -> TensorLikeType:
+    assert isinstance(a, TensorLikeType) or isinstance(b, TensorLikeType)
+
+    # If b is a number. mypy doesn't like isinstance(b, NumberType)
+    if not isinstance(b, TensorLikeType):
+        if b == 1.0:
+            return a  # type: ignore[return-value]
+        elif b == 2.0:
+            return a * a  # type: ignore[return-value]
+        elif b == 0.5:
+            return torch.sqrt(a)  # type: ignore[arg-type]
+        elif b == 1.0 / 3.0:
```

Comment:
Nice addition

Reply:
But unfortunately incorrect when `a` is negative.

For mathematical details see: https://www.intmath.com/blog/mathematics/is-cube-root-the-same-as-raising-to-power-13-10460

There's no `torch.cbrt` and `prims.cbrt` is currently implemented as:
```
def _cbrt_aten(a: torch.Tensor):
    return pow(a, (1 / 3))
```
which is also incorrect for negative values and doesn't match NumPy or C.

The correct implementation for the `cbrt` function in terms of `pow` would be
```py
def cbrt(x):
    return -((-x)**(1/3)) if x < 0 else x**(1/3)
```

```py
def cbrt(x):
    return -((-x)**(1/3)) if x < 0 else x**(1/3)
    
import numpy as np
a = -2.0
print(np.cbrt(a)) # -1.259921049894873
print(a ** (1 / 3)) # (0.6299605249474367+1.0911236359717214j)
print(cbrt(a)) # -1.2599210498948732
print(np.power(a, 1 / 3)) # nan

import torch
print(torch.tensor(a), 1 / 3)) # tensor(nan)
```

In Python 3.11 `cbrt` is added to the `math` library https://github.com/python/cpython/commit/ac867f10b49322e25f34d2d8abd8e63c86834750.

Reply:
wtf.

From [`np.power`](https://numpy.org/doc/stable/reference/generated/numpy.power.html):
>Negative values raised to a non-integral value will return nan. To get complex results, cast the input to complex, or specify the dtype to be complex (see the example below).

Inherited from [`std::pow`](https://en.cppreference.com/w/cpp/numeric/math/pow):
> pow(base, exp) returns NaN and raises [FE_INVALID](https://en.cppreference.com/w/cpp/numeric/fenv/FE_exceptions) if base is finite and negative and exp is finite and non-integer.

Now, after thinking about this for quite some time, this behaviour, although very counterintuitive, makes sense. 

tl;dr. `cbrt` is defined for real numbers as "the usual cubic root", while `pow(a, z)` is defined as `exp(zlog(a))` where `log(a)` represents the main branch of the complex logarithm. The function `z^{1/3}` has 3 possible values that are always arranged on a triangle of the form:
![image](https://user-images.githubusercontent.com/3291265/176157024-6e60b57f-ac3e-4500-a0d8-15e4ad0bfb35.png)
The definition `exp(zlog(a))` happens to be choosing the first root (counterclockwise) in that diagram starting at 1. This happens for all negative numbers, and it's annoying, as the real root of `z^{1/3}` is the second one. Note that, for higher roots like `z^{1/4}` or `z^{1/5}` we get similar diagrams to this one but with a square, a pentagon, etc. In general, you'd need to choose a very particular branch of the logarithm to get the real root, which would make for a very odd casing for negative numbers in `std::pow`. As such, the  C++ committee settled for this behaviour which, although weird, is correct and efficient.

So, from here we get a few things:
- I'll remove that `cbrt` from there
- We need to fix `cbrt`
- We need to give good definitions for what we mean with `sqrt` (this one's easy), `pow` and `cbrt`.

Reply:
fwiw, I summarised these points in a small comment in https://github.com/pytorch/pytorch/pull/80443. Hope that works. Thank you for pointing this out @IvanYashchuk!